### PR TITLE
CBG-465: Create lazy marshal function

### PR DESCRIPTION
**Pull request by JRascagneres**

### CODE REVIEW

**Comment by adamcfraser** [comment]
We should be in the habit of writing a meaningful commit message, at least for the main commit.  That would get pulled up into the PR description as a bonus.

File: db/document.go [hunk 1]
```diff
@@ -167,29 +167,31 @@ func NewDocument(docid string) *Document {
...
-// TODO: review whether this can just return raw body when available
-func (doc *Document) MarshalBody() ([]byte, error) {
-	marshalled, err := json.Marshal(doc.Body())
-	if err != nil {
-		return []byte{}, pkgerrors.Wrapf(err, "Error marshalling JSON")
+func (doc *Document) BodyBytes() ([]byte, error) {
+	if doc._rawBody == nil && doc._body != nil {
+		bodyBytes, err := json.Marshal(doc._body)
+		if err != nil {
+			err = pkgerrors.Wrapf(err, "Error marshalingl document body bytes")
```
Comment (adamcfraser) [inline db/document.go]: Typo in marshalling, and should probably omit 'bytes'.

File: db/document.go [hunk 2]
```diff
@@ -167,29 +167,31 @@ func NewDocument(docid string) *Document {
...
+			return nil, err
```
Comment (adamcfraser) [inline db/document.go]:
```suggestion
			return nil, pkgerrors.Wrapf(err, "Error marshalling document body")
```
Can probably streamline this code as the above.

File: db/document.go [hunk 3]
```diff
@@ -401,12 +403,13 @@ func (doc *Document) IsSGWrite(rawBody []byte) (isSGWrite bool, crc32Match bool)
 	}
 
 	// Since raw body isn't available, marshal from the document to perform body hash comparison
-	docBody, err := doc.MarshalBody()
+
```
Comment (adamcfraser) [inline db/document.go]: Can remove this whitespace

**Review by adamcfraser — CHANGES_REQUESTED** [review]
A few style suggestions/typos - otherwise looks good.

**Review by bbrks — APPROVED** [review]

**Pull request merged**